meson.build

```
@@ -26,6 +26,7 @@ rt = cc.find_library('rt')
 pipewire = dependency('libpipewire-0.3', version: '>= 0.3.2')
 wayland_client = dependency('wayland-client')
 wayland_protos = dependency('wayland-protocols', version: '>=1.14')
+iniparser = cc.find_library('iniparser')
```

Is there no pkg-config file for this library?

---

Sadly [no](https://bugs.archlinux.org/task/45589?string=iniparser&project=1&type%5B0%5D=&sev%5B0%5D=&pri%5B0%5D=&due%5B0%5D=&reported%5B0%5D=&cat%5B0%5D=&status%5B0%5D=&percent%5B0%5D=&opened=&dev=&closed=&duedatefrom=&duedateto=&changedfrom=&changedto=&openedfrom=&openedto=&closedfrom=&closedto=).

---

Fails on FreeBSD (and likely other BSDs) as packages are not in the default search path of base compiler in order to avoid tainting build environment. See also https://wiki.freebsd.org/WarnerLosh/UsrLocal#Library_paths

```
$ meson setup _build
[...]
meson.build:29:0: ERROR: C shared or static library 'iniparser' not found

$ pkg info -l iniparser
iniparser-4.1:
        /usr/local/include/dictionary.h
        /usr/local/include/iniparser.h
        /usr/local/lib/libiniparser.a
        /usr/local/lib/libiniparser.so
        /usr/local/lib/libiniparser.so.1
        /usr/local/share/licenses/iniparser-4.1/LICENSE
        /usr/local/share/licenses/iniparser-4.1/MIT
        /usr/local/share/licenses/iniparser-4.1/catalog.mk
```

Maybe try the following
```suggestion
iniparser = cc.find_library('iniparser', dirs: [join_paths(get_option('prefix'), get_option('libdir'))])
```

---

This should ne a change you submit to meson, not to each and every project using it.

---

See also https://github.com/mesonbuild/meson/issues/7746